src/views/virtualmachines/actions/components/CloneVMModal/components/CloningStatus.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC, useEffect } from 'react';
+
+import { modelToGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import VirtualMachineCloneModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineCloneModel';
+import { V1alpha1VirtualMachineClone } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  AlertVariant,
+  ProgressStep,
+  ProgressStepper,
+  ProgressStepVariant,
+  Stack,
+} from '@patternfly/react-core';
+
+import { CLONING_STATUSES, STATUS_TO_PROGRESS_VARIANT } from '../utils/constants';
+
+type CloningStatusProps = {
+  onComplete: (status: CLONING_STATUSES) => void;
+  vmCloneRequest: V1alpha1VirtualMachineClone;
+};
+
+const CloningStatus: FC<CloningStatusProps> = ({ onComplete, vmCloneRequest }) => {
+  const { t } = useKubevirtTranslation();
+
+  const [freshVMCloneRequest] = useK8sWatchResource<V1alpha1VirtualMachineClone>(
+    vmCloneRequest && {
+      groupVersionKind: modelToGroupVersionKind(VirtualMachineCloneModel),
+      name: vmCloneRequest.metadata.name,
+      namespace: vmCloneRequest.metadata.namespace,
+    },
+  );
+
+  const cloneRequest = freshVMCloneRequest || vmCloneRequest;
+
+  const phrase = cloneRequest?.status?.phase;
+
+  useEffect(() => {
+    if (
+      [CLONING_STATUSES.FAILED, CLONING_STATUSES.SUCCEEDED, CLONING_STATUSES.UNKNOWN].includes(
+        freshVMCloneRequest?.status?.phase as CLONING_STATUSES,
+      )
+    ) {
+      onComplete(freshVMCloneRequest?.status?.phase as CLONING_STATUSES);
```

Comment:
why onComple running here? and not in parent?
this component should only show the status and no for logic here, the parent can handle the logic

Reply:
I thought it was better to have everything here instead of putting more logic in the parent.
The parent already have other kind of logics and it seems everything messed up a little bit. But if you think is better move everything on the parent, okay.